spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -159,7 +160,15 @@ public Table loadTable(Identifier ident, String version) throws NoSuchTableExcep
           sparkTable.snapshotId() == null,
           "Cannot do time-travel based on both table identifier and AS OF");
 
-      return sparkTable.copyWithSnapshotId(Long.parseLong(version));
+      try {
```

Comment:
I'm not sure this can come up, but do we allow version tags to be SnapshotIds? 

Like can I tag snapshot 2 to be known as 1?

Weird edge case so I don't think we really need to handle it, just thinking if this is a potential issue with the lookup code here

Reply:
Currently there's no restrictions on what references can be named. For the lookup code, since the ref is in  I think we should always be able to differentiate between snapshot ID and ref since for refs it will be in a quoted identifier, and should always fail the Long.parseLong(). 

But that's just me reading the code, I think it's worth having a unit test just for this case to give us that confidence that it works as expected in this scenario. cc @jackye1995 let me know your thoughts

Reply:
Currently there's no restrictions on what references can be named. For the lookup code, since the ref is in  I think we should always be able to differentiate between snapshot ID and ref since for refs it will be in a quoted identifier, and should always fail the Long.parseLong(). 

But that's just me reading the code, I think it's worth having a unit test just for this case to give us that confidence that it works as expected in this scenario. cc @jackye1995 let me know your thoughts

Reply:
I added a test case specifically for this. Unlike Trino, Spark directly ignores the type of the `VERSION AS OF`, so if a tag name matches exactly the snapshot ID, then snapshot ID is always chosen.

I think this is a okay limitation, because people can work around it by adding some text like `snapshot-123456890` as the tag name. But we should make it very clear in documentation.